### PR TITLE
Reorganize Readme

### PR DESCRIPTION
**Pull request by slotThe**
So this is a first draft of some of the ideas I have regarding #512; I
thought writing this "in the open" would be much better, as it's a
documentation change and lots of people will probably have lots of
different views here.  Also it's much easier to check GitHubs rendering
of `org-mode` this way :)

Done so far:

  - `readme.md` was converted to org-mode (I have not extensively
    checked the GitHub rendering yet), some minor changes and updates
    (e.g. I fixed some broken links).
  - There's now also some instructions for installing xmobar via system
    package managers.
  - The documentation of the plugins has been moved to `doc/plugins.org`
  - We now have a `contributing.org`, which details not only how to
    contribute to xmobar, but also how to write your own plugin.
  - The quick start part of the readme now resides in
    `doc/quick-start.org`.

I have not touched the website script at all yet.

### CODE REVIEW

File: doc/quick-start.org [hunk 1]
```diff
@@ -0,0 +1,361 @@
+Xmobar can either be configured using the configuration language, or
```
Comment (jaor) [inline doc/quick-start.org:1]: note for the future: this is fine as a first version, but possibly too detailed for a "quick start".  an idea might be to extract something shorter.  but perhaps that's splitting hairs too thinly :)

**Review by jaor — COMMENTED** [review]
this looks pretty good to me, thanks! i've added a few comments that are just nits, or notes for future improvement... if you agree with the "contributing" split (maybe the "contributing section could simply go to the "readme.org", contributing.org be renamed to something like "plugins.org"), and moving the detailed compilation sections away from the top page, maybe we could do that in this PR.

File: contributing.org [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+problems you are having, as well as the proposed solution.
+
+** Making the Change
+
+It's best to create a separate branch in your clone of the [[https://github.com/jaor/xmobar/][xmobar repo]]
+and then push this branch to your fork.
+
+Please also *never* merge =master= into the branch you have created.
+=git= offers the very neat option of rebasing your changes on top of
+another branch; use that instead.
```
[the remark this answers is not in the document]
Comment (slotThe) [reply contributing.org:18]: This is perhaps worded a bit weirdly; what I mean is when someone creates a new branch (in their fork), does some changes, merges master into their newly created branch (still in their fork), and then creates a pull request.  This is a bit of a mess everytime it happens, though thankfully it does not happen all that often.  I can remove this altogether if you want

File: contributing.org [hunk 3]
```diff
@@ -0,0 +1,117 @@
+* Contributing Changes/Patches to xmobar
+
+Want to contribute to xmobar? You've come to the right place! This
+document will help guide you in this endeavour.
+
+In case you want to make any non-trivial change to xmobar, it's always
+best to talk with the community first. Currently the best way to do that
+is to create an issue on GitHub. There, you can talk through the
```
[the remark this answers is not in the document]
Comment (slotThe) [reply contributing.org]: Done

File: contributing.org [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+other documentation that your pull request touches upon.  For example,
+if you add a new plugin you should update the [[./doc/plugins.org][plugins documentation]].
+
+* Writing Your Own Plugin
+
+Writing a plugin for xmobar is very simple!
+
+First, you need to create a data type with at least one constructor.
+Next you must declare this data type an instance of the =Exec= class, by
+defining the 1 needed method (alternatively =start= or =run=) and 3
```
[the remark this answers is not in the document]
Comment (slotThe) [reply contributing.org]: Done

File: doc/plugins.org [hunk 5]
```diff
@@ -0,0 +1,1547 @@
...
+
+There are two types of arguments: ones that all monitors share (the so
+called /default monitor arguments/) and arguments that are specific to a
+certain monitor.
+
+All Monitors accept a common set of arguments, described below in
+[[Default Monitor Arguments]]. Some monitors also accept additional options
+that are specific to them. When specifying the list of arguments in your
+configuration, the common options come first, followed by =--=, followed
+by any monitor-specific options.
```
[the remark this answers is not in the document]
Comment (slotThe) [reply doc/plugins.org]: Done

**Comment by slotThe** [comment]
Thanks for the review!  I've fixed the small issues for now.  

I like your suggestions regarding the structural improvements and will implement this as soon as I have some free time :)

**Comment by jaor** [comment]
On Mon, Dec 14 2020, slotThe wrote:

> +Please also *never* merge =master= into the branch you have created.
> +=git= offers the very neat option of rebasing your changes on top of
> +another branch; use that instead.
>
> This is perhaps worded a bit weirdly; what I mean is when someone
> creates a new branch (in their fork), does some changes, merges master
> into their newly created branch (still in their fork), and then
> creates a pull request. This is a bit of a mess everytime it happens,
> though thankfully it does not happen all that often. I can remove this
> altogether if you want

but one has to rebase against master in case anything changes
there... maybe we could mention that?

**Comment by slotThe** [comment]
On Mon, Dec 14 2020 06:05, jao wrote:
> but one has to rebase against master in case anything changes
> there... maybe we could mention that?

That's what I originally meant with

  > git offers the very neat option of rebasing your changes on top of
  > another branch; use that instead

I'll remove the merging stuff and just put a note in there that rebasing
exists and one should probably be prepared to do so if the pr goes
through some revisions

**Comment by jaor** [comment]
On Mon, Dec 14 2020, slotThe wrote:

> Thanks for the review! I've fixed the small issues for now.

excellent.  thanks for your work!

> I like your suggestions regarding the structural improvements and will
> implement this as soon as I have some free time :)

sounds good.  shall we merge this PR in the meantime?

**Comment by slotThe** [comment]
On Mon, Dec 14 2020 06:32, jao wrote:
> sounds good.  shall we merge this PR in the meantime?

Sure, that sounds like a good idea; I'll convert the draft to a proper pr

**Review by jaor — APPROVED** [review]

**Pull request merged**